python/lsst/scarlet/lite/io.py

```diff
@@ -0,0 +1,598 @@
+        data_shallow_copy["shape"] = tuple(data["shape"])
+        psf_shape = data_shallow_copy.pop("psf_shape")
+        data_shallow_copy["psf_center"] = tuple(data["psf_center"])
+        data_shallow_copy["psf"] = np.array(data["psf"]).reshape(psf_shape).astype(dtype)
+        data_shallow_copy["sources"] = {
+            int(bid): ScarletSourceData.from_dict(source, dtype=dtype)
+            for bid, source in data["sources"].items()
+        }
+        return cls(**data_shallow_copy)
+
+
+class ScarletModelData:
+    """A container that propagates scarlet models for an entire catalog."""
+
+    def __init__(self, bands: list[Any], psf: np.ndarray, blends: dict[int, ScarletBlendData] | None = None):
+        """Initialize an instance
+
+        Parameters
+        ----------
+        bands:
+            The names of the bands.
+            The order of the bands must be the same as the order of
+            the multiband model arrays, and SEDs.
+        psf:
+            The 2D array of the PSF in scarlet model space.
+            This is typically a narrow Gaussian integrated over the
+            pixels in the exposure.
+        blends:
+            Map from parent IDs in the source catalog
+            to scarlet model data for each parent ID (blend).
+        """
+        self.bands = bands
+        self.psf = psf
+        if blends is None:
+            blends = {}
+        self.blends = blends
+
+    def json(self) -> str:
+        """Serialize the data model to a JSON formatted string
+
+        Returns
+        -------
+        result : `str`
+            The result of the object converted into a JSON format
+        """
+        result = {
+            "bands": self.bands,
+            "psfShape": self.psf.shape,
+            "psf": list(self.psf.flatten().astype(float)),
+            "blends": {bid: blend.as_dict() for bid, blend in self.blends.items()},
+        }
+        return json.dumps(result)
+
+    @classmethod
+    def parse_obj(cls, data: dict) -> ScarletModelData:
+        """Construct a ScarletModelData from python decoded JSON object.
+
+        Parameters
+        ----------
+        data:
+            The result of json.load(s) on a JSON persisted ScarletModelData
+
+        Returns
+        -------
+        result:
+            The `ScarletModelData` that was loaded the from the input object
+        """
+        data_shallow_copy = dict(data)
+        model_psf = (
+            np.array(data_shallow_copy["psf"]).reshape(data_shallow_copy.pop("psfShape")).astype(np.float32)
+        )
+        data_shallow_copy["psf"] = model_psf
+        data_shallow_copy["blends"] = {
+            int(bid): ScarletBlendData.from_dict(blend) for bid, blend in data["blends"].items()
+        }
+        return cls(**data_shallow_copy)
+
+
+class ComponentCube(Component):
+    """Dummy component for a component cube.
+
+    This is duck-typed to a `lsst.scarlet.lite.Component` in order to
```

Comment:
The is primarily to allow users converting custom models into a persistable format. So if, for example, a user creates a class that is not separable into spectral and morphological components, they can export the full 3D `(bands, y, x)` model and save it. It can then be loaded in a blend, just not updated because it has no information about the parameters used to fit the model.